content/blog/2021/04/2021-04-16-deprecated-ruby-runtime.adoc

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: "Deprecated ruby-runtime and associated plugins"
+tags:
+- ruby
+- plugins
+authors:
+- halkeye
+opengraph:
+  image: /images/post-images/2021-04-deprecating-ruby-runtime.png
+---
+Approximately
+https://github.com/jenkinsci/ruby-runtime-plugin/commit/d368b087fadb3282c9b537f0fa6d9a150b080c73[10 years ago, ruby-runtime was first released]. It was an experiment to let
+plugins to be written in ruby but still get integrated into the main
+java Jenkins process.
+
+Over the years though, the experiment has stopped being maintained, the
+dozen or so plugins that use it take a lot of user effort to make work,
+and the Jenkins ruby runtime does not support Java 11.
+Specifically the last few years have been really hard on the Jenkins
+core developers, as they still have to maintain all the hacks and
+processes to keep the ruby runtime barely working.
+
+One month from now, May 15th, 2021, the Jenkins update center will stop distributing
+ruby-runtime and its https://plugins.jenkins.io/ruby-runtime/#dependencies[associated plugins].
+
+== What does this mean for you?
+
+If you are one of the few users who are using the following plugins. The
+only change will come for new installs. Suspended plugins can stay
+installed, but can not be newly installed without manually downloading
+releases.
+
+The lists below provide additional information about the most popular plugins based on the Ruby runtime.
```

Comment:
The list is a combination of alternatives and justification.

When we know an alternative (especially for a plugin like the gitlab-hooks plugin with over 20 000 installations), I think it will help the reader to understand the alternatives for that plugin.  I've seen at least two tables to divs bug reports recently that included ruby-runtime and dependent plugins.  Offering alternatives when we have them.

The justification part was a sanity check for me.  I was worried that one or more of the plugins might still be active, so I checked the last release date and offered it as part of the text.  I don't have strong objections if that justification is removed.  Collecting the data has already met my needs.

Reply:
We can remove the headings, but I think the list of individual plugins is a really good idea. Most people won't click through.

Reply:
I used the headings as an excuse to visually separate the most frequently installed plugins from the others.  I'd like to have the most popular plugins at the top of the list, but don't feel a strong need for headings.  

The headings and the partitions they represent are arbitrary and not really relevant to readers.  If they aren't using a plugin, they don't care how many installations it has.  If they are using a plugin, they still don't care how many installations it has.